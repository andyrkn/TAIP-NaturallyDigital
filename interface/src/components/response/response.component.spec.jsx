import React from "react";
import renderer from "react-test-renderer";
import Response from "./response.component";
import { assert } from "chai";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({
    adapter: new Adapter()
});

describe("Response component", () => {

    let payload;

    beforeAll(() => {
        payload = {
            attribute1: "value1",
            attribute2: "value2"
        }
    })

    it("makes a snapshot", () => {
        const tree = renderer.create(<Response />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("renders the institution and request type", () => {
        const wrapper = shallow(<Response payload={payload} />);
        const sections = wrapper.find("span");
        const attribute1 = sections.getElements()[0];
        const value1 = sections.getElements()[1];
        const attribute2 = sections.getElements()[2];
        const value2 = sections.getElements()[3];

        assert.lengthOf(sections, 4);
        assert.equal(attribute1.props.children, Object.keys(payload)[0]);
        assert.equal(attribute2.props.children, Object.keys(payload)[1]);
        assert.equal(value1.props.children[1], payload.attribute1);
        assert.equal(value2.props.children[1], payload.attribute2);
    })
});