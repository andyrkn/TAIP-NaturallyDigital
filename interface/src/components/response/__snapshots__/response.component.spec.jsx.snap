// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response component makes a snapshot 1`] = `
<div
  id="response"
>
  <h3>
    Response
  </h3>
</div>
`;
